core/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -30,36 +30,30 @@
 
     private static final DeprecationLogger DEPRECATION_LOGGER = new DeprecationLogger(Loggers.getLogger(ParseField.class));
 
-    private final String camelCaseName;
     private final String underscoreName;
     private final String[] deprecatedNames;
     private String allReplacedWith = null;
 
     public ParseField(String value, String... deprecatedNames) {
-        camelCaseName = Strings.toCamelCase(value);
         underscoreName = Strings.toUnderscoreCase(value);
-        if (deprecatedNames == null || deprecatedNames.length == 0) {
-            this.deprecatedNames = Strings.EMPTY_ARRAY;
-        } else {
-            final HashSet<String> set = new HashSet<>();
-            for (String depName : deprecatedNames) {
-                set.add(Strings.toCamelCase(depName));
-                set.add(Strings.toUnderscoreCase(depName));
-            }
-            this.deprecatedNames = set.toArray(new String[set.size()]);
+        final HashSet<String> set = new HashSet<>();
+        set.add(Strings.toCamelCase(value));
```

Comment:
it is a bit odd that if the underscore case version is equal to the camelCase version (think "term"), we always add it to the list of deprecated names. This doesn't seem to have side effects apart from returning [term, term] from `getAllNamesIncludedDeprecated` (something that would be fixed in master I think) but maybe we should prevent adding it to the deprecated ones for correctness?